inginious/frontend/plugins/task_hints/pages/user_hint_manager.py

```diff
@@ -0,0 +1,134 @@
+
+from inginious.frontend.parsable_text import ParsableText
+
+class UserHintManager(object):
+
+    """
+        Manage the user hints information in database, as the hints
+        that was unlocked, it's individual penalty, and cumulative
+        penalty, to be applied in the student's final grade for the
+        respective task.
+    """
+
+    def __init__(self, username, task_id, database):
+
+        self._username = username
+        self._task_id = task_id
+        self._database = database
+
+    def check_locked_hint_status(self, hints):
+
+        """ Method to check each hint status, and return the content
+            if it is allowed
+        """
+        user_hints = self._database.user_hints.find_one({"taskid":  self._task_id,
+                                            "username":  self._username})
+        if user_hints is None:
+            user_hints = self.create_hint_list_for_user()
+            return 200, ""
+
+        self.update_allowed_hints_list(user_hints, hints)
+
+        to_show_hints = {}
+        allowed_hints = user_hints["allowedHints"]
+        for key in hints:
```

Comment:
I think this for is not very understandable, why  do you iterate over the key when you just want to access the hint itself? you can do something like `for key, hint in hints.items()` or `for hint in hints.values()` in case you just want to access the hint's data  thus you do not need to access every hint like `hints[key][another key]`.